aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -46,28 +46,21 @@ static void apply_batched_inverse_lib(Tensor& self, Tensor& self_inv, Tensor& in
 
   auto& allocator = *::c10::cuda::CUDACachingAllocator::get();
 
-  if (use_loop_launch(batch_size, n)) {
-    int* p_infos = infos.data_ptr<int>();
-    auto main_stream = at::cuda::getCurrentCUDAStream();
-
-    at::cuda::CUDAEvent main_event;
-    main_event.record(main_stream);
-
-    for (int64_t i = 0; i < batch_size; i++) {
-      auto stream = at::cuda::getStreamFromPool();
-      at::cuda::CUDAStreamGuard guard(stream);
+  // heuristic:
+  //   cublas_x_batched doesn't work very well for small batchsize
+  //   cublas_x_batched is intended to be used for matrices of small sizes where the launch overhead is a significant factor.
+  const bool use_cusolver_parallel_launch = (batch_size <= 8) || (/* batch_size > 8 && */ n >= 512);
 
-      main_event.block(stream);
+  if (use_cusolver_parallel_launch) {
```

Comment:
Nice to see this simplified. So you're confident that the multistreaming doesn't improve performance?